src/dataflow/src/render/reduce.rs

```diff
@@ -1063,6 +1066,135 @@ where
         })
 }
 
+/// Accumulates values for the various types of accumulable aggregations.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+enum Accumulator {
+    /// Accumulates boolean values.
+    Bool {
+        /// The number of `true` values observed.
+        trues: isize,
+        /// The number of `false` values observed.
+        falses: isize,
+        /// The total number of values observed.
+        total: isize,
+    },
+    /// Accumulates the count of non-NULL values.
+    Count {
+        /// The number of non-NULL values observed.
+        non_nulls: isize,
+        /// The total number of values observed.
+        ///
+        /// This value is not actually observed directly by the code in this
+        /// module, but it serves to ensure that `is_zero()` returns false when
+        /// at least one non-NULL value has been observed. This is required to
+        /// produce an explicit `0` when a group contains zero elements;
+        /// otherwise differential would produce no results at all.
+        total: isize,
+    },
+    /// Accumulates simple numeric values.
+    SimpleNumber {
+        /// The sum of all non-NULL values observed.
+        sum: i128,
+        /// The number of non-NULL values observed.
+        count: isize,
+    },
+}
+
+impl Default for Accumulator {
+    fn default() -> Accumulator {
+        Accumulator::Count {
+            non_nulls: 0,
+            total: 0,
+        }
+    }
```

Comment:
I think this is fine because DD doesn't use `default()`, but if this accidentally gets used as a zero, it would panic the addition unless it were changed to first check `.is_zero()`. No issue that I can see, but wanted to flag.

Reply:
Ack, restructured with an explicit `Accumulator::Zero` type that can be added to any other type.